### PR TITLE
Implement the `I` element in version 3

### PR DESCRIPTION
**Pull request by Lyxal**

### CODE REVIEW

File: shared/src/main/scala/Elements.scala [hunk 1]
```diff
@@ -448,6 +448,21 @@ object Elements:
...
+      "I",
+      "Interleave",
+      List("interleave"),
+      "a: lst, b: lst -> Interleave a and b"
+    ) { case (a, b) =>
+      val temp = ListHelpers.interleave(
+        ListHelpers.makeIterable(a),
+        ListHelpers.makeIterable(b)
+      )
+      if a.isInstanceOf[String] && b.isInstanceOf[String] then temp.mkString("")
```
Comment (ysthakur) [inline shared/src/main/scala/Elements.scala:462]:
```suggestion
      if a.isInstanceOf[String] && b.isInstanceOf[String] then temp.mkString
```
`mkString` uses `""` by default

File: shared/src/main/scala/ListHelpers.scala [hunk 2]
```diff
@@ -39,6 +39,23 @@ object ListHelpers:
...
+    var mutLeft = left.toList
+    var mutRight = right.toList
+    while mutLeft.nonEmpty && mutRight.nonEmpty do
+      out += mutLeft.head
+      out += mutRight.head
+      mutLeft = mutLeft.tail
+      mutRight = mutRight.tail
+
+    out ++= mutLeft
+    out ++= mutRight
```
Comment (ysthakur) [inline shared/src/main/scala/ListHelpers.scala:53]:
```suggestion
    val leftIter = left.iterator
    val rightIter = right.iterator
    while leftIter.hasNext && rightIter.hasNext do
      out += leftIter.next()
      out += rightIter.next()

    out ++= leftIter
    out ++= rightIter
```
I'm not sure if those last two lines will still work with an iterator, but they can definitely be modified a bit to work

File: shared/src/main/scala/Elements.scala (hunk 1 above)
Comment (Lyxal) [reply shared/src/main/scala/Elements.scala]: I blame copilot

File: shared/src/main/scala/ListHelpers.scala (hunk 2 above)
Comment (Lyxal) [reply shared/src/main/scala/ListHelpers.scala:53]: Once again, I blame copilot

**Review by ysthakur — APPROVED** [review]

**Pull request merged**